File: js/advanced/4c_remove_duplicates_from_sorted_array.js
//Remove duplicates from sorted array and return no. of unique values in it.

var removeDuplicates = function(nums) {
    let l = 0
    for(let r=1; r<nums.length; r++){
        if(nums[r] !== nums[l]){
            nums[l+1] = nums[r]
            l++
        }
    }
    return l+1
};

File: c++/advanced/4c_remove_duplicates_from_sorted_array.cpp
//Remove duplicates from sorted array and return no. of unique values in it.

int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int i=0;
        for(int j=1; j<n; j++){
            if(nums[j] != nums[i]){
                nums[i+1] = nums[j];
                i++;
            }
        }
        return i+1;
    }